docs/src/tutorial.md

```diff
@@ -0,0 +1,241 @@
+# Tutorial
+
+In this tutorial we can see the main uses of `SolverBenchmark`.
+
+First, let's create fake data. It is imperative that the data for each solver is stored
+in `DataFrame`s, and the collection of different solver must be stored in a dictionary of
+`Symbol` to `DataFrame`.
+
+In our examples we'll use the following data.
+
+```@example ex1
+using DataFrames, Printf, Random
+
+Random.seed!(0)
+
+n = 10
+names = [:alpha, :beta, :gamma]
+stats = Dict(name => DataFrame(:id => 1:n,
+         :name => [@sprintf("prob%03d", i) for i = 1:n],
+         :status => map(x -> x ? :success : :failure, rand(n) .< 0.75),
+         :f => randn(n),
+         :t => 1e-3 .+ rand(n) * 1000,
+         :iter => rand(10:10:100, n),
+         :irrelevant => randn(n)) for name in names)
+```
+
+The data consists of a (fake) run of three solvers `alpha`, `beta` and `gamma`.
+Each solver has a column `id`, which is necessary for joining the solvers (names
+can be repeated), and columns `name`, `status`, `f`, `t` and `iter` corresponding to
+problem results. There is also a column `irrelevant` with irrelevant information.
+
+Here is solver `alpha`:
+
+```@example ex1
+stats[:alpha]
+```
+
+## Tables
+
+The first thing we may want to do is produce a table for each solver. Notice that the
+solver result is already a DataFrame, so there are a few options available in other
+packages, as well as simply printing the DataFrame.
+Our concern here is two-fold: producing publication-ready LaTeX tables, and web-ready
+markdown tables.
+
+The simplest use is `foo_table(io, dataframe)`. Here is printout to the `stdout`.
+
+```@example ex1
+using SolverBenchmark
+
+markdown_table(stdout, stats[:alpha])
+```
+
+```@example ex1
+latex_table(stdout, stats[:alpha])
+```
+
+Alternatively, you can print to a file.
+
+```@example ex1
+open("alpha.tex", "w") do io
+  println(io, "\\documentclass[varwidth=20cm,crop=true]{standalone}")
+  println(io, "\\usepackage{longtable}")
+  println(io, "\\begin{document}")
+  latex_table(io, stats[:alpha])
+  println(io, "\\end{document}")
```

Comment:
Should we have a function for this?

Reply:
I'm not sure it will be in high demand. I think we should wait and see.